packages/components/src/slot-fill/use-force-update.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+export default function useForceUpdate() {
+	const [ , setState ] = useState( {} );
+	return useCallback( () => setState( {} ), [] );
```

Comment:
I've also seen `useReducer` used for a similar pattern, removing the need for `useCallback`.

Example from memory: I haven't actually tested it.

```js
const [ , forceUpdate ] = useReducer( () => {}, {} );
```